opm/input/eclipse/Schedule/Well/Well.cpp

```diff
@@ -614,63 +618,79 @@ bool Well::updateWellGuideRate(double guide_rate_arg) {
 }
 
 
-bool Well::updateFoamProperties(std::shared_ptr<WellFoamProperties> foam_properties_arg) {
+bool Well::updateFoamProperties(std::shared_ptr<WellFoamProperties> foam_properties_arg)
+{
     if (this->wtype.producer()) {
-        throw std::runtime_error("Not allowed to set foam injection properties for well " + name()
-                                 + " since it is a production well");
+        throw std::runtime_error {
+            fmt::format("Not allowed to set foam injection properties for well {}, "
+                        "since it is a production well.", this->name())
+        };
     }
+
     if (*this->foam_properties != *foam_properties_arg) {
-        this->foam_properties = foam_properties_arg;
+        this->foam_properties = std::move(foam_properties_arg);
         return true;
     }
 
     return false;
 }
 
-
-bool Well::updatePolymerProperties(std::shared_ptr<WellPolymerProperties> polymer_properties_arg) {
+bool Well::updatePolymerProperties(std::shared_ptr<WellPolymerProperties> polymer_properties_arg)
+{
     if (this->wtype.producer()) {
-        throw std::runtime_error("Not allowed to set polymer injection properties for well " + name() +
-                                 " since it is a production well");
+        throw std::runtime_error {
+            fmt::format("Not allowed to set polymer injection properties for well {}, "
+                        "since it is a production well.", this->name())
+        };
     }
+
     if (*this->polymer_properties != *polymer_properties_arg) {
-        this->polymer_properties = polymer_properties_arg;
+        this->polymer_properties = std::move(polymer_properties_arg);
         return true;
     }
 
     return false;
 }
 
-bool Well::updateMICPProperties(std::shared_ptr<WellMICPProperties> micp_properties_arg) {
+bool Well::updateMICPProperties(std::shared_ptr<WellMICPProperties> micp_properties_arg)
+{
     if (this->wtype.producer()) {
-        throw std::runtime_error("Not allowed to set micp injection properties for well " + name() +
-                                 " since it is a production well");
+        throw std::runtime_error {
+            fmt::format("Not allowed to set micp injection properties for well {}, "
```

Comment:
Setting micp injection properties for well {} is not allowed, since it is a production well